test/java/org/opendatakit/briefcase/model/form/FormKeyTest.java

```diff
@@ -27,4 +29,18 @@ public void regression_wrong_form_name_when_creating_keys_from_briefcase_form_de
     FormKey key2 = FormKey.from(new FormStatus(BriefcaseFormDefinition.resolveAgainstBriefcaseDefn(formFile.toFile(), false, briefcaseFolder.toFile())));
     assertThat(key1, is(key2));
   }
+
+  @Test
+  public void different_and_same_key_check_if_equal() {
```

Comment:
Here are my thoughts about this test:
- `FormKey` is a value object, so we should assert whether an instance **is** or **is not** another instance. 
  Language, in this case, is important because in Java, being equal to something and "is"-ing  that something implies different equality rules.
  In this case we want to verify that objects are effectively the same, as if they were primitive values.
- I wouldn't bother asserting the `null` case
- The mixed types case is formally wrong because you're only matching it against a String, which doesn't probe it won't be equal to any other infinite types that you could use for that assertion. I'd remove that assertion.
- Also, in this codebase we use Hamcrest matchers with `assertThat`. it's important to follow conventions and coding standards. In this case, I'd expect to have `assertThat` assertions instead of `assertFalse` or `assertTrue`.
- Another remark about coding is the mix of statically imported `assertTrue` versus the fully qualified `Assert.assertFalse`. I'd expect everything statically imported, or fully qualified. Not a mix.